MTEB

Muennighoff opened the pull request.

stevhliu commented on mteb.md:
````diff
@@ -0,0 +1,138 @@
+python mteb_meta.py results/average_word_embeddings_komninos
+```
+
+This will produce a `mteb_metadata.md` file looking like:
````
```suggestion
The script will produce a `mteb_metadata.md` file that looks like this:
```

stevhliu commented on mteb.md:
````diff
@@ -0,0 +1,138 @@
+---
+```
+
+Now add the metadata to the top of a README.md of any model on the hub like [here](https://huggingface.co/Muennighoff/SGPT-5.8B-weightedmean-msmarco-specb-bitfit/blob/main/README.md) and it will show up on the [leaderboard](https://huggingface.co/spaces/mteb/leaderboard) after refreshing!
````
```suggestion
Now add the metadata to the top of a README.md of any model on the Hub, like this [SGPT-5.8B](https://huggingface.co/Muennighoff/SGPT-5.8B-weightedmean-msmarco-specb-bitfit/blob/main/README.md) model, and it will show up on the [leaderboard](https://huggingface.co/spaces/mteb/leaderboard) after refreshing!
```

stevhliu reviewed (approved):
Really cool, I love how easy y'all made it to benchmark a model! 🚀

I would consider adding a brief background on why embeddings are useful and even compare them to previous embedding benchmarks. Then you can point out where the previous benchmarks are lacking and introduce MTEB and how it addresses these issues. It doesn't have to be anything too long, and I think readers will definitely be interested in learning more!

It might also be cool to embed the leaderboard in the blog but it's kind of long. What do you think?

Muennighoff commented:
> Really cool, I love how easy y'all made it to benchmark a model! 🚀
> 
> I would consider adding a brief background on why embeddings are useful and even compare them to previous embedding benchmarks. Then you can point out where the previous benchmarks are lacking and introduce MTEB and how it addresses these issues. It doesn't have to be anything too long, and I think readers will definitely be interested in learning more!
> 
> It might also be cool to embed the leaderboard in the blog but it's kind of long. What do you think?

Amazing feedback, thanks a lot! I'm looking into the first two points.

I think embedding the leaderboard is a great idea - do you have an example blog you can link to where someone has embedded a HuggingFace space?

stevhliu commented:
Yeah check out this [blog](https://github.com/huggingface/blog/blob/main/playlist-generator.md) by @NimaBoscarino !

Muennighoff commented:
> Yeah check out this [blog](https://github.com/huggingface/blog/blob/main/playlist-generator.md) by @NimaBoscarino !

That's really nice! I think indeed our leaderboard would be too long though 😁

I added a section to explain text embeddings for now. There's a more detailed comparison with previous benchmarks in the paper linked at the top 👍

stevhliu reviewed (approved):
Thanks for adding more context about text embeddings, looks good to me now! I think if it looks ok locally, it should be fine. Lastly, remember to update the publish date :)

NimaBoscarino commented:
> Yeah check out this [blog](https://github.com/huggingface/blog/blob/main/playlist-generator.md) by @NimaBoscarino !

Just a heads up that embedding Gradio Spaces with web components is preferred over using iframes now! I'd use this as a better reference: https://github.com/huggingface/blog/blob/main/spaces_3dmoljs.md
